sui-execution/latest/sui-adapter/src/programmable_transactions/context.rs

```diff
@@ -107,191 +108,166 @@ mod checked {
-                // so to mimic this "off limits" behavior, we act as if the coin has less balance than
-                // it really does
-                let Some(Value::Object(ObjectValue {
-                    contents: ObjectContents::Coin(coin),
-                    ..
-                })) = &mut gas.inner.value
-                else {
-                    invariant_violation!("Gas object should be a populated coin")
-                };
-
-                let max_gas_in_balance = gas_charger.gas_budget();
-                let Some(new_balance) = coin.balance.value().checked_sub(max_gas_in_balance) else {
-                    invariant_violation!(
-                        "Transaction input checker should check that there is enough gas"
-                    );
-                };
-                coin.balance = Balance::new(new_balance);
-                gas
-            } else {
-                InputValue {
-                    object_metadata: None,
-                    inner: ResultValue {
-                        last_usage_kind: None,
-                        value: None,
-                    },
-                }
-            };
-            let native_extensions = new_native_extensions(
-                state_view.as_child_resolver(),
-                input_object_map,
-                !gas_charger.is_unmetered(),
-                protocol_config,
-                metrics.clone(),
-                tx_context.epoch(),
-            );
-
-            // Set the profiler if in CLI
-            #[skip_checked_arithmetic]
-            move_vm_profiler::tracing_feature_enabled! {
-                use move_vm_profiler::GasProfiler;
-                use move_vm_types::gas::GasMeter;
-
-                let tx_digest = tx_context.digest();
-                let remaining_gas: u64 =
-                    move_vm_types::gas::GasMeter::remaining_gas(gas_charger.move_gas_status())
-                        .into();
-                gas_charger
-                    .move_gas_status_mut()
-                    .set_profiler(GasProfiler::init(
-                        &vm.config().profiler_config,
-                        format!("{}", tx_digest),
-                        remaining_gas,
-                    ));
-            }
+    /// A `LinkedContext` is an execution context with a specific linkage that has been determined.
+    /// This is usually for a specific command, but the linkage can be derived, e.g., for the
+    /// entire PTB.
+    pub struct LinkedContext<'ctx, 'vm, 'state, 'a> {
+        /// The execution context
+        pub ctx: &'ctx mut ExecutionContext<'vm, 'state, 'a>,
+        pub vm_instance: MoveVM<'state>,
+        /// The specified linkage for this linked context.
+        /// This is a mapping of runtime_id -> storage_id
+        pub linkage: ResolvedLinkage,
+        /// A "reverse" linkage of storage_id -> runtime_id
+        pub reverse_linkage: BTreeMap<ObjectID, ObjectID>,
```

Comment:
Why isn't this a part of ResolvedLinkage?

Reply:
No particular reason, I think adding in to that would make sense to me.